### DISCUSSION TITLE
Is there any way to decouple the storage from the postgres cluster?

### DISCUSSION THREAD

**Discussion by JacksonZ03** [cloudnative-pg/cloudnative-pg#4278] [Q&A] [answered]
I'm currently trying to create a cluster following the defaults laid out in the documentation, but upon testing I found that specifying the storageclass automatically creates the PVC and the PV, but then when I delete it, the PVC and PV also disappears with it.

So to prevent this, I specified Retain in the storageclass configuration, but since I'm using Longhorn to manage storage, once you delete a PVC, the previously associated PV turns from Bound to Released, rather than to Available.
This means if I recreate the postgres cluster, a new set of PVCs will be created along with a new set of PVs as well. The data that was held in the previous set of PVs remain hanging.

Right now, for the rest of my application, I have kept the PVC resource defintions separate to the actual services and deployments. This is so that even when I delete an application, the PVC stays around and I can easily restart the application and it will automatically reuse the existing PVC.

Is there any way for me to do something similar with CloudNative-PG as well?

Or at least make it so that when I delete the cluster, the storage doesn't get deleted with it but is instead kept around and reused if I recreate the cluster from the manifest definition?

**Comment by sxd** [comment]
Hi @JacksonZ03 

Have you tried the hibernate command with the plugin ?

Regards,

**Reply by JacksonZ03** [reply]
Ah, that's neat. 
Is it also possible to resume the cluster by recreating the cluster by running this command?
```
kubectl apply -f postgres-cluster.yaml
```

Or do you have to resume it with `kubectl cnpg hibernate off <cluster-name>`?

**Reply by sxd** [reply, marked as answer]
Hi,

Probably you want to take a look here https://cloudnative-pg.io/documentation/current/declarative_hibernation/ which is the declarative way and here https://cloudnative-pg.io/documentation/current/kubectl-plugin/#cluster-hibernation which is the one using the plugin

Regards,

**Reply by JacksonZ03** [reply]
What's the difference between running `kubectl apply -f postgres-cluster.yaml` after it's in hibernation and `kubectl cnpg hibernate off <cluster-name>`? Are they just different ways of applying the same thing, or is there a practical difference between the two ways?

If I hiberate the cluster and then unhibernate it by running `kubectl cnpg hibernate on  <cluster-name>` and then `kubectl cnpg hibernate off  <cluster-name>`, is that the same as restarting the whole thing? In other words, if I change something in the original manifest, would it pick up the change and apply those changes automatically when I run `kubectl cnpg hibernate off <cluster-name>`?

**Reply by sxd** [reply]
Hi @JacksonZ03 

With the first one you are probably deleting the annotation and with the second one you're just setting that annotation to off

Closing this discussion as it was already answered

Regards!
